### DISCUSSION TITLE
Querying a large list of items with multiple filtering options

### DISCUSSION THREAD

**Discussion by hornta** [reduxjs/redux-toolkit#2113] [Q&A]
Hi!

We have this large list of items with multiple types of filters we can apply on the items. I'm thinking of making it so the endpoint can take a bunch of parameters like "name", "category", "assigne", "owner", much like the GitHub issues list. However I'm a little bit worried that all the different kinds of filtering combinations would get cached in the client with every filter change causing the client to hold an enormous amount of data in the end. 

We don't yet have pagination so we will get around 500 items for each query. Would it make sense to just clear the cache for every change to the filters? Am I overthinking this?

**Comment by phryneas** [comment]
Once your UI is not using a cache entry any more (e.g. by starting a new search) for a certain amount of time (default 60 sec), it will be removed from the cache. 
You can configure that timeout using [keepUnusedDataFor](https://redux-toolkit.js.org/rtk-query/api/createApi#keepunuseddatafor-1) to something shorter if you are concerned, but I honestly would not be too much. That is probably still in the kilobytes or low megabytes - compare that to the amount of images your site uses.
